File: src/de/schauderhaft/cleanCode/exercise4/LabelGenerator.java
package de.schauderhaft.cleanCode.exercise4;

import javax.swing.JLabel;

import de.schauderhaft.cleanCode.stubs.SrvParameterCache;
import de.schauderhaft.cleanCode.stubs.StringStatic;
import de.schauderhaft.cleanCode.stubs.Translator;

public class LabelGenerator {

    JLabel stageInfoLabel;

    public JLabel getLblStageInfo() {
        if (stageInfoLabel == null) {
            stageInfoLabel = new JLabel();

            stageInfoLabel.setText(getLabelText());
            stageInfoLabel.setVisible(getVisibility());
        }

        return stageInfoLabel;
    }

    private String getLabelText() {

        String text = getI18nText("lblStageInfoPrd");

        try {
            String importUrl = SrvParameterCache
                    .getParameterAsString(SrvParameterCache.WEB_URL_DEFAULT);
            if (!StringStatic.isEmptyOrNullValue(importUrl)) {
                importUrl = importUrl.toLowerCase();

                if (isTestInstance(importUrl)) {
                    text = getI18nText("lblStageInfoTest");
                } else if (isQaInstance(importUrl)) {
                    text = getI18nText("lblStageInfoQs");
                }
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return text;
    }

    private boolean getVisibility() {
        boolean visibility = true;
        try {
            String importUrl = SrvParameterCache
                    .getParameterAsString(SrvParameterCache.WEB_URL_DEFAULT);
            if (!StringStatic.isEmptyOrNullValue(importUrl)) {
                importUrl = importUrl.toLowerCase();
                if (isProductionInstance(importUrl)) {
                    visibility = false;
                }
            }
        } catch (Exception e) {
            visibility = false;
        }
        return visibility;
    }

    private boolean isProductionInstance(String url) {
        return !(isTestInstance(url) || isQaInstance(url));
    }

    private boolean isQaInstance(String url) {
        return url.contains("grglmrpfqs") || url.contains("gru23c12")
                || url.contains("sysglmqs") || url.contains("sysgmrpf-qs");
    }

    private boolean isTestInstance(String url) {
        return url.contains("grglmrpft") || url.contains("sysmrpt")
                || url.contains("gru23b12") || url.contains("gru23b13");
    }

    private String getI18nText(String string) {

        return Translator.translate(string);
    }
}

File: src/de/schauderhaft/cleanCode/exercise4/LabelGenerator.cs
using System;
using System.Windows.Forms;
using de.schauderhaft.cleanCode.stubs;

namespace de.schauderhaft.cleanCode.exercise4
{
    public class LabelGenerator 
    {
        private Label stageInfoLabel;
        
        public Label GetLblStageInfo() 
        {
            if (stageInfoLabel == null)
            {
                stageInfoLabel = new Label();
                
                stageInfoLabel.Text = GetLabelText();
                stageInfoLabel.Visible = GetVisibility();
            }
            return stageInfoLabel;
        }

        private string GetLabelText()
        {
            string text = GetI18nText("lblStageInfoPrd");
        
            try
            {
                string importUrl = SrvParameterCache
                    .GetParameterAsString(SrvParameterCache.WEB_URL_DEFAULT);
                if (!StringStatic.IsEmptyOrNullValue(importUrl))
                {
                    importUrl = importUrl.ToLower();
                    
                    if (IsTestInstance(importUrl)) 
                    {
                        text = GetI18nText("lblStageInfoTest");
                    }
                    else if (IsQaInstance(importUrl)) 
                    {
                        text = GetI18nText("lblStageInfoQs");
                    }
                }
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
                Console.WriteLine(e.StackTrace);
            }
            return text;
        }
        
        private bool GetVisibility()
        {
            bool visibility = true;
            try
            {
                string importUrl = SrvParameterCache
                        .GetParameterAsString(SrvParameterCache.WEB_URL_DEFAULT);
                if (!StringStatic.IsEmptyOrNullValue(importUrl))
                {
                    importUrl = importUrl.ToLower();
                    if (IsProductionInstance(importUrl)) 
                    {
                        visibility = false;
                    }
                }
            }
            catch 
            {
                visibility = false;
            }
            return visibility;
        }
        
        private bool IsProductionInstance(string url)
        {
            return !(IsTestInstance(url) || IsQaInstance(url));
        }
        
        private bool IsQaInstance(string url)
        {
            return url.Contains("grglmrpft")
                    || url.Contains("sysmrpt")
                    || url.Contains("gru23b12")
                    || url.Contains("gru23b13");
        }
        
        private bool IsTestInstance(string url)
        {
            return url.Contains("grglmrpfqs")
                    || url.Contains("gru23c12")
                    || url.Contains("sysglmqs")
                    || url.Contains("sysgmrpf-qs");
        }
        
        private string GetI18nText(string message)
        {
            return Translator.Translate(message);
        }
    }
}